File: src/MessageHash/MessageHashAPI.cpp
#include <iostream>
#include <MessageHashAPI.h>
#include <MessageHash.h>

#include <memory>

MESSAGE_HASH_RETURN_TYPE HashMsgSHA256 (const std::string& msg)
    {
         MessageHash hashMess;
         hashMess.HashSha256 (msg);
#ifdef __EMSCRIPTEN__
        return hashMess.HashHex().c_str(); 
#else
         return hashMess.HashHex ();
#endif         
    }

MESSAGE_HASH_RETURN_TYPE HashMsg (const std::string& msg, const std::string& hashContext)
    {
         MessageHash hashMess;
         hashMess.Hash(msg,hashContext);
#ifdef __EMSCRIPTEN__
        return hashMess.HashHex().c_str() ; 
#else         
        return hashMess.HashHex ();
#endif        
    }

MESSAGE_HASH_RETURN_TYPE ListHashFunc ()
    {
        MessageHash hashMess;
#ifdef __EMSCRIPTEN__
        return hashMess.ListAvailableHash().c_str() ; 
#else        
        return hashMess.ListAvailableHash () ; 
#endif        
    }

File: src/MessageHash/MessageHashAPI.js

function HashMsgSHA256API (argHash){
  const hashsha256func = Module.cwrap('HashMsgSHA256','string',['string']);
  var res = hashsha256func(argHash);
  return res;
}
  
function HashMsg (argHash, argHashFunc){
  const hashfunc = Module.cwrap('HashMsg','string',['string']);
  var res = hashfunc(argHash, argHashFunc);
  return res;
}

function ListHashFunc(){
  const listhashfunc = Module.cwrap('ListHashFunc','string',[]);
  var res = listhashfunc();
  return res; 
}
